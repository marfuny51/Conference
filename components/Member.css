.Table {
    text-align: center;
}

.TableMember {
    border-collapse: collapse;
    width: 70vw;
    height: 10vh;
    margin: 6vh auto 2vh auto;
}

.TableMemberAdd {
    border-collapse: collapse;
    width: 70vw;
    height: 10vh;
    margin: 6vh auto 2vh auto;
    opacity:0.3;
}

.TableHead {
    color: #00D970;
}

.TableHead th {
    font-family: 'Muli', sans-serif;
    font-size: 1.2vw;
    background-color: #127a16;
    color: white;
    letter-spacing: 0.1vw;
    padding: 1vw;
    border: 0vw;
}

.TableBody tr {
    height: 6vh;
    background-color: #d3f1d4;
    border-bottom: 0.15vw solid #127a16;
}

.TableBody tr:hover {
    background-color: #f5f5f5;
}

.TableBody tr td {
    font-family: 'Muli', sans-serif;
    font-size: 1.1vw;
    border: 0vw;
    padding: 0.5vw;
}

.Read {
    font-family: 'Muli', sans-serif;
    font-size: 1vw;
    height: 4.2vh;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
}

.Read:hover {
    font-weight: bold;
    transform: scale(1.03);
    border:0.2vw solid #456835;
    cursor: pointer;
}

.Read2 {
    font-family: 'Muli', sans-serif;
    font-size: 1vw;
    height: 4.2vh;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
    margin-right: 1.2vw;
}

.Read2:hover {
    font-weight: bold;
    transform: scale(1.03);
    border:0.2vw solid #456835;
    cursor: pointer;
}

.Add {
    font-family: 'Muli', sans-serif;
    font-size: 1.1vw;
    font-weight: bold;
    height: 6.5vh;
    background-color: #a2eca3;
    color: black;
    border-radius: 0.5vw;
    border:0.2vw solid #77b45b;
}

.Add:hover {
    transform: scale(1.03);
    border:0.2vwsolid #456835;
    cursor: pointer;
}

.NavLink {
    color: #172c11;
    font-weight: bold;
}

.LoadData {
    font-size: 1.2vw;
    text-align: center;
    color: white;
}

.AddMember {
    width: 48vw;
    height: 32vh;
    position: absolute;
    top: 35%;
    left: 26%;
    background-color: #d3f1d4;
    border: 0.2vw solid #77b45b;
    border-radius: 0.5vw;
}

.Title {
    font-family: 'Muli', sans-serif;
    font-weight: bold;
    letter-spacing:0.1vw;
    padding-top: 1vw;
}

.Input {
    font-family: 'Muli', sans-serif;
    font-size: 1.1vw;
    text-align: left;
    padding: 3.8vh 1vw 3vh 4vw;
}

.Input input{
    width: 9vw;
    height: 2.5vh;
}

.Valide {
    color: red;
}
