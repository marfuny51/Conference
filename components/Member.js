import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from './events';

import './Member.css';

class Member extends React.PureComponent {

  static propTypes = {
    member:PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      phone: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
      job: PropTypes.string.isRequired,
    }),
  };

  deleteMember = () => {
    let question = confirm('Do you sure? You are going to delete this information.');
    if(question) {
      voteEvents.emit('EDeleteMember', this.props.member.id);
    }
  }

  editMember = () => {
    voteEvents.emit('EEditMember', this.props.member.id);
  }

  render() {
    
    return (
      <tr key={this.props.member.id}>
        <td>{this.props.member.name}</td>
        <td>{this.props.member.phone}</td>
        <td>{this.props.member.email}</td>
        <td>{this.props.member.job}</td>
        <td><input type="button" value="Edit" className='Read' disabled = {(this.props.mode===8)?true:false} onClick={this.editMember}/></td>
        <td><input type="button" value="Delete" className='Read' disabled = {(this.props.mode===8)?true:false} onClick={this.deleteMember}/></td>
      </tr>
    );

  }

}

export default Member;
