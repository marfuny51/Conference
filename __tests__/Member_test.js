import React from 'react';
import Member from '../components/Member';
import renderer from 'react-test-renderer';

test('работа Member', () => {

    let member={id:7265,name:"Anna Beresovnik",phone:"+375294587965",email:"berezovnik@gmail.com",job:"Student"};
    // создаём тестовую версию компонента
    const component = renderer.create(
      <Member 
      member={member}
    />
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
  
    // найдём в вёрстке компонента кнопку 'Edit'
    const buttonEdit = component.root.findByProps( {value:'Edit'}); 
    // и "нажмём" на неё
    buttonEdit.props.onClick();
    window.confirm = () => {true};
 
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
  
    // найдём в вёрстке компонента кнопку 'Delete'
    const buttonDelete = component.root.findByProps( {value:'Delete'}); 
    // и "нажмём" на неё
    buttonDelete.props.onClick();
  
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    
    buttonEdit.props.onClick(); 
    
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    
    buttonDelete.props.onClick();

    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    /*
    // можно эмулировать события, передавая в качестве объекта события то что нам нужно:
    wrapper.find('select').simulate('change', {
      target: { value: "hello" },
    });
    */
  
  });
  