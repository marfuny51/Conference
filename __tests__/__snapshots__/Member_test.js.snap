// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Member 1`] = `
<tr>
  <td>
    Anna Beresovnik
  </td>
  <td>
    +375294587965
  </td>
  <td>
    berezovnik@gmail.com
  </td>
  <td>
    Student
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Member 2`] = `
<tr>
  <td>
    Anna Beresovnik
  </td>
  <td>
    +375294587965
  </td>
  <td>
    berezovnik@gmail.com
  </td>
  <td>
    Student
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Member 3`] = `
<tr>
  <td>
    Anna Beresovnik
  </td>
  <td>
    +375294587965
  </td>
  <td>
    berezovnik@gmail.com
  </td>
  <td>
    Student
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Member 4`] = `
<tr>
  <td>
    Anna Beresovnik
  </td>
  <td>
    +375294587965
  </td>
  <td>
    berezovnik@gmail.com
  </td>
  <td>
    Student
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;

exports[`работа Member 5`] = `
<tr>
  <td>
    Anna Beresovnik
  </td>
  <td>
    +375294587965
  </td>
  <td>
    berezovnik@gmail.com
  </td>
  <td>
    Student
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Edit"
    />
  </td>
  <td>
    <input
      className="Read"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Delete"
    />
  </td>
</tr>
`;
